main.cpp

```diff
@@ -58,46 +58,86 @@ Purpose:  This project will show you the difference between member functions and
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(int v, const char* identifier) //1
+    {
+        value = v; // 2
+        name = identifier; // 3
+    }
+
+    int value;
+    std::string name;
 };
 
-struct <#structName1#>                                //4
+struct X // 4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) // 5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if (a && b)
```

Comment:
Be more explicit when you check for a nullptr
`if( a != nullptr && b != nullptr )
- [ ] revise